sqlx-sqlite/src/connection/mod.rs

```diff
@@ -557,6 +611,64 @@ impl LockedSqliteHandle<'_> {
         let ret = unsafe { sqlite3_get_autocommit(self.as_raw_handle().as_ptr()) };
         ret == 0
     }
+
+    /// Sets the soft heap limit for the current thread.
+    ///
+    /// This function sets a soft limit on the amount of heap memory that can be allocated by SQLite.
+    /// The limit is in bytes. If `limit` is zero, the heap limit is disabled.
+    ///
+    /// Returns the previous heap limit. If the heap limit was never set, returns 0.
+    ///
+    /// See: https://www.sqlite.org/c3ref/hard_heap_limit64.html
+    pub fn soft_heap_limit(&mut self, limit: i64) -> i64 {
+        unsafe { sqlite3_soft_heap_limit64(limit) }
+    }
+
+    /// Retrieves statistics about a database connection.
+    ///
+    /// This function is used to retrieve runtime status information about a single database connection.
+    /// The `status` parameter determines which statistic to retrieve.
+    ///
+    /// Returns a tuple containing `(current_value, highest_value_since_reset)`.
+    /// If `reset` is true, the highest value is reset to the current value after retrieval.
+    ///
+    /// See: https://www.sqlite.org/c3ref/db_status.html
+    pub fn db_status(
+        &mut self,
+        status: SqliteDatabaseStatus,
+        reset: bool,
+    ) -> Result<(i32, i32), Error> {
+        let mut current = 0i32;
+        let mut highest = 0i32;
+
+        let result = unsafe {
+            sqlite3_db_status(
+                self.as_raw_handle().as_ptr(),
+                status.into(),
+                &mut current,
+                &mut highest,
+                if reset { 1 } else { 0 },
+            )
+        };
+
+        if result == SQLITE_OK {
+            Ok((current, highest))
+        } else {
+            Err(self.guard.handle.expect_error().into())
+        }
+    }
+
+    /// Attempts to free as much heap memory as possible from the database connection.
+    ///
+    /// This function causes SQLite to release some memory used by the database connection,
+    /// such as memory used to cache prepared statements.
+    ///
+    /// Returns the number of bytes of memory released.
+    ///
+    /// See: https://www.sqlite.org/c3ref/release_memory.html
```

Comment:
This links to the wrong page. It should be: https://www.sqlite.org/c3ref/db_release_memory.html